manta-pay/src/crypto/key.rs

```diff
@@ -27,12 +28,17 @@ use manta_util::{impl_empty_codec, into_array_unchecked};
 #[derive(Clone, Copy, Debug, Default, Eq, Hash, Ord, PartialEq, PartialOrd)]
 pub struct Blake2sKdf;
 
-impl KeyDerivationFunction for Blake2sKdf {
-    type Key = [u8];
-    type Output = [u8; 32];
+impl SecretKeyType for Blake2sKdf {
+    type SecretKey = Vec<u8>;
+}
+
+impl PublicKeyType for Blake2sKdf {
+    type PublicKey = [u8; 32];
+}
 
+impl Derive for Blake2sKdf {
```

Comment:
This is not technically a key agreement scheme so it should not use this API, it should use the other KDF one.